lib/worker-pipeline.js

```diff
@@ -57,7 +58,15 @@ class AssetComputeWorkerPipeline {
             return this.transformers;
         }
 
-        const transformer = new WorkerCallbackTransformer(renditionCallback, this.options.manifest);
+        //const transformer = new WorkerCallbackTransformer(renditionCallback, this.options.manifest);
+        let transformer;
+        if (this.options.manifest.hasBatchMode || this.options.manifest.inputs.hasBatchMode) {
```

Comment:
We already know here that the call is in batch mode. As you got here from `batchWorker` call in `api.js` (line 72).
So adding an api/method to `AssetComputeWorkerPipeline` will help you decouple from the `hasBatchMode` check and also you won't need to add this field in Manifest(which it seems does not belong there)

Reply:
I don't totally agree, since the manifest is supposed to describe the worker's capabilities, and "batch" is one of those capabilities.